Sources/App/Core/Search.swift

```diff
@@ -175,7 +183,18 @@ enum Search {
             .column(keywords)
             .column(hasDocs)
             .column(nullInt, as: levenshteinDist)
+            .column(ts_rank(vector: tsvector, query: tsquery), as: tsrankvalue)
             .from(searchView)
+            .from(plainto_tsquery(mergedTerms), as: tsquery)
+            .from(
+                TextSearch.toVector(
+                    concat(with: " ",
+                        coalesce(packageName, emptyString),
+                        coalesce(summary, emptyString),
+                        arrayToString(keywords, delimiter: " ")
+                    ),
+                    weight: .a
```

Comment:
Actually, I don't think we need to use `setweight` at all. We're only applying weights A and B in distinct subqueries which don't actually mix. I.e. we weight the vector in the package (1) and the author (2) subqueries with A and the keywords (3) subquery with B.

However, these results aren't actually displayed in the same list and so their respective weightings have no effect:

![CleanShot 2022-08-07 at 16 56 01@2x](https://user-images.githubusercontent.com/65520/183297158-3975cc58-e374-446e-94f6-d3c61b1bd4d9.png)

It does not harm but we should remove that to simplify the SQL a bit to keep it a little more readable.

Reply:
We don't need the weights. Originally I thought I'd be trying to mix different weights, but wasn't able to work out how to do so outside of the SELECT inline setup. End result - didn't end up using it. If we care about the specific rank values (for example, to fuse in Score as a modifier), then it might be worthwhile, but for this round I think we can absolutely strip it out.

Reply:
I'm pretty sure we can fix the joining issue you encountered. Are there some cases we know would benefit from using weights? Might be something to schedule for a follow-up. It's definitely valuable to have had it around so we can pull it from git history should we need it!

Reply:
Since I didn't get it working, I don't have any hard numbers for what benefit it could provide. My gut was telling me to rank keywords and package titles above anything in the summary content based on past experiences, but I don't know that we'd see much benefit from even doing that right now.